lib/views/frontend/spree/checkout/payment/_stripe_apple_pay.html.erb

```diff
@@ -64,6 +64,15 @@
     form.appendChild(hiddenInput);
   };
 
+  function hideApplePayRadioButtonForNonAppleDevices() {
+    var isSafari = !!navigator.userAgent.match(/Version\/[\d\.]+.*Safari/);
+    var isiOS = /iPad|iPhone|iPod/.test(navigator.userAgent) && !window.MSStream;
+
+    if (!isSafari && !isiOS) {
+      $('#payment-method-fields li.radio:contains(Apple Pay)').hide()
```

Comment:
```suggestion
      $('#payment-method-fields li.radio:contains(Apple Pay)').hide()
```

let's use `Spree.stripeApplePayPaymentMethod` to find this radio button not regexp